Update use cases for 'local' packages (#4048)
fishtown-analytics/docs.getdbt.com: website/docs/docs/build/packages.md

</File>

### Local packages

> A "local" package is another dbt project that you have access to from within the local file system. It can be installed by specifying the path to the project. The best-supported pattern is when the project is nested within a subdirectory, relative to the current project's directory.

Reviewer: ```suggestion
A "local" package is a dbt project accessible from your local file system. You can install it by specifying the project's path. It works best when you nest the project within a subdirectory relative to your current project's directory.
```

<File name='packages.yml'>

```yaml
packages:
  - local: relative/path/to/subdirectory
```

</File>

Other patterns may work in some cases, but not always. For example, if you install this project as a package elsewhere, or try running it on a different system, the relative and absolute paths will yield the same results.

<File name='packages.yml'>

```yaml
packages:

>   # not recommended - the support for these patterns vary

Reviewer: ```suggestion
  # not recommended - support for these patterns vary
```

>   # not recommended - these support for these patterns vary

Reviewer: did you mean 'the'?
```suggestion
  # not recommended - the support for these patterns vary
```

  - local: /../../redshift   # relative path to a parent directory
  - local: /opt/dbt/redshift # absolute path on the system
```

</File>

> As such, there are a few specific use cases where a "local" package is recommended:

Reviewer: ```suggestion
There are a few specific use cases where we recommend using a "local" package:
```

> 1. A monorepo containing multiple projects, each nested in a subdirectory. "Local" packages enable combining projects together, for coordinated development and deployment.

Reviewer: sugg having some bolded categories so it's pretty clear from the get go

```suggestion
1. **Monorepo** &mdash; When you have multiple projects, each nested in a subdirectory, within a monorepo. "Local" packages allow you to combine projects for coordinated development and deployment.
```

> 2. Testing changes to one project/package within the context of a downstream project/package that uses it. By temporarily switching the installation to a "local" package, you can make changes to the former and immediately test them in the latter, enabling quicker iteration. This is similar to [editable installs](https://pip.pypa.io/en/stable/topics/local-project-installs/) in Python.

Reviewer: ```suggestion
2. **Testing changes** &mdash; To test changes in one project or package within the context of a downstream project or package that uses it. By temporarily switching the installation to a "local" package, you can make changes to the former and immediately test them in the latter for quicker iteration. This is similar to [editable installs](https://pip.pypa.io/en/stable/topics/local-project-installs/) in Python.
```

> 3. A nested project that defines fixtures and tests for a project of utility macros — for example, [the integration tests within the `dbt-utils` package](https://github.com/dbt-labs/dbt-utils/tree/main/integration_tests)

Reviewer: ```suggestion
3. **Nested project** &mdash; When you have a nested project that defines fixtures and tests for a project of utility macros, like [the integration tests within the `dbt-utils` package](https://github.com/dbt-labs/dbt-utils/tree/main/integration_tests).
```
